File: matlab_code/oldcode/homework2_1.m
%Çó¸ùºÅ(a^2+b^2)=p

N=10;%N=3
a=-2.2;%e200
b=3.4;

a=abs(a);
b=abs(b);
p=max(a,b);
q=min(a,b);
for i =1:N
    r=(q/p)^2;
    s=r/(4+r);
    p=p+2*s*p;
    q=s*q;
end
q
p

File: anacondacode/tensorflow_environment/old_code/计算方法/homework2_1.py
# -*- coding: utf-8 -*-
"""
Created on Fri Sep 13 14:09:51 2019

@author: LJZ
"""

#求(根号a)+(根号b)=p

N=10
a=2.2
b=3.4

p=max(a,b)
q=min(a,b)
for i in(1,N+1):
    r=(q/p)**2
    s=r/(4+r)
    p=p+2*s*p
    q=s*q
print(q)
print(p)